src/ui/targetingTree.ts

```diff
@@ -36,11 +37,18 @@ const comparatorMap = {
 export const renderTargetingTree = (
     featureConfigs: FeatureConfig[],
     environments: Environment[],
-    variations: Variation[]
+    variations: Variation[],
+    audiences: AudienceSchema[]
 ) => {
     const targetingTree = ux.tree()
-
-    featureConfigs.forEach((config) => {
+    let configs = featureConfigs
+    // if there's audience match filter, we want to build audience name map
+    // and replace audience ids in feature configs with audience names
+    if (configs.some((config) => hasAudienceMatchFilter(config))) {
+        const audienceNameMap = buildAudienceNameMap(audiences)
+        configs = replaceAudienceIdsInFeatureConfigs(configs, audienceNameMap)
+    }
```

Comment:
could we not save the iteration of checking for an audience match filter and replacing the audience ids in the feature config by just building and passing the audience map to `insertRulesTree`? since it's already handling replacing the values if they exist

Reply:
I tried it out that way and it felt cumbersome having another a function argument be passed on to multiple levels of nested calls, all while being optional. 

This way the configs get changed at the top and the nested functions stay untouched

Reply:
Ah, it looks like `insertRulesTree` still has the param but it's not being used. The main advantage I'd see is that we'd avoid having `replaceAudienceIdsInFeatureConfigs` duplicating the logic of traversing the feature config

Reply:
I think it would be easier if all of these methods were encapsulated in some sort of class instead, as it would help prevent then need to pass around these parameters in such a nested way and have it available on the instance. I can make a ticket for it, as I would argue the duplication of traversing is worth it for now to decrease the amount of arguments we need to pass down

Reply:
I agree that a class instance would be a better way to store a reference to the associated envs/variations/audiences

Reply:
I still think this is traversing the config two more times than it needs to. I'd advocate we pass the param to `insertRulesTree` and then create a follow up ticket to convert this to a class